addons/godot-xr-tools/player/player_body.gd

```diff
@@ -399,6 +392,39 @@ func override_player_height(key, value: float = -1.0):
 	var override = _player_height_overrides.values().min()
 	_player_height_override = override if override != null else -1.0
 
+# Estimate body forward direction
+func _estimate_body_forward_dir() -> Vector3:
+	var forward = Vector3()
+	var camera_basis : Basis = camera_node.global_transform.basis
+	var camera_forward : Vector3 = -camera_basis.z;
+
+	if camera_forward.y > 0.75:
+		# User is looking up
+		forward = up_player_plane.project(-camera_basis.y).normalized()
+	elif camera_forward.y < -0.75:
+		# User is looking down
+		forward = up_player_plane.project(camera_basis.y).normalized()
+	else:
+		forward = up_player_plane.project(camera_forward).normalized()
+
+	if (left_hand_node and left_hand_node.get_is_active()
+		and right_hand_node and right_hand_node.get_is_active()
+		and body_forward_mix > 0.0):
+		# See if we can mix in our estimated forward vector based on controller position
+		# Note, in Godot 4.0 we should check tracker confidence
+
+		var tangent = right_hand_node.global_transform.origin - left_hand_node.global_transform.origin
+		tangent = (tangent * Vector3(1.0, 0.0, 1.0)).normalized()
```

Comment:
This math appears to assume the player is in the global Y+/UP frame; where-as the player may be walking around a sphere or on a wall.

You may want to consider something like:
```gdscript
    tangent = up_player_plane.project(tangent).normalized()
    var hands_forward = up_player_vector.cross(tangent).normalized()
```

Reply:
Good catch!

Reply:
Ok, added this and the other changes you suggested below, should indeed fix the orientation issues on the spherical world